pilot/pkg/networking/core/v1alpha3/listener_builder.go

```diff
@@ -200,59 +199,56 @@ func (builder *ListenerBuilder) buildVirtualInboundListener(env *model.Environme
 
 func (builder *ListenerBuilder) patchListeners(push *model.PushContext) {
 	if builder.node.Type == model.Router {
-		envoyfilter.ApplyListenerPatches(networking.EnvoyFilter_GATEWAY, builder.node, push, builder.gatewayListeners, false)
+		envoyfilter.ApplyListenerPatches(networking.EnvoyFilter_GATEWAY, builder.node, push, builder.gatewayListeners)
 		return
 	}
 
-	patchOneListener := func(listener *xdsapi.Listener) *xdsapi.Listener {
+	patchOneListener := func(listener *xdsapi.Listener, ctx networking.EnvoyFilter_PatchContext) *xdsapi.Listener {
 		if listener == nil {
 			return nil
 		}
 		tempArray := []*xdsapi.Listener{listener}
-		tempArray = envoyfilter.ApplyListenerPatches(networking.EnvoyFilter_SIDECAR_OUTBOUND, builder.node, push, tempArray, true)
+		tempArray = envoyfilter.ApplyListenerPatches(ctx, builder.node, push, tempArray)
 		// temp array will either be empty [if virtual listener was removed] or will have a modified listener
 		if len(tempArray) == 0 {
 			return nil
 		}
 		return tempArray[0]
 	}
-	builder.virtualListener = patchOneListener(builder.virtualListener)
-	builder.virtualInboundListener = patchOneListener(builder.virtualInboundListener)
-	builder.managementListeners = envoyfilter.ApplyListenerPatches(networking.EnvoyFilter_SIDECAR_INBOUND, builder.node,
-		push, builder.managementListeners, true)
+	builder.virtualListener = patchOneListener(builder.virtualListener, networking.EnvoyFilter_SIDECAR_OUTBOUND)
+	builder.virtualInboundListener = patchOneListener(builder.virtualInboundListener, networking.EnvoyFilter_SIDECAR_INBOUND)
```

Comment:
I realized another issue with skipAdd. I added it also to handle the patchOneListener case. Since we removed skipAdd, I think its easier to add the virtualListener to outbound, virtualInbound listener to inbound and simply concatenate both. Alternatively, we can restore the skipAdd because pathOneListener will need skipAdd = true